File: 03-02-2025/Using Enum for Basic Calculator with Error Handling.c
#include <stdio.h>

typedef enum {
    SUCCESS, ERROR
} Status;

Status divide(int a, int b, float *result) {
    if (b == 0) return ERROR;
    *result = (float)a / b;
    return SUCCESS;
}

int main() {
    int num1 = 10, num2 = 0;
    float result;

    if (divide(num1, num2, &result) == ERROR) {
        printf("Error! Division by zero.\n");
    } else {
        printf("Result: %.2f\n", result);
    }

    return 0;
}

File: 03.02.2025/Using Enum for Basic Calculator with Error Handling.py
from enum import Enum

class Status(Enum):
    SUCCESS = 1
    ERROR = 2

def divide(a, b):
    if b == 0:
        return Status.ERROR, None
    return Status.SUCCESS, a / b

status, result = divide(10, 0)
if status == Status.ERROR:
    print("Error! Division by zero.")
else:
    print(f"Result: {result}")
